datastore/db/bytebase/FertiScan/schema_0.0.12.sql

```diff
@@ -287,6 +288,26 @@ IF (EXISTS (SELECT 1 FROM information_schema.schemata WHERE schema_name = 'ferti
     FOR EACH ROW
     EXECUTE FUNCTION update_fertilizer_timestamp();
 
+    -- Trigger function for the `inspection` table
+    CREATE OR REPLACE FUNCTION update_inspection_original_dataset_protection()
+    RETURNS TRIGGER AS $$
+    BEGIN
+    IF (TG_OP = 'UPDATE') AND (OLD.original_dataset IS NULL) THEN
```

Comment:
is (TG_OP = 'UPDATE') check needed? Because the trigger is configured with `BEFORE UPDATE`.

Reply:
Yes but in case the function is called else where this add a layer of protection without impacting much efficiency